ccdproc/combiner.py

```diff
@@ -124,14 +124,15 @@ def weights(self):
         return self._weights
 
     @weights.setter
-    def weights(self, value):
+    def weights(self, value, axis=0):
         if value is not None:
             if isinstance(value, np.ndarray):
-                if value.shape == self.data_arr.data.shape:
-                    self._weights = value
-                else:
-                    raise ValueError(
-                        "dimensions of weights do not match data.")
+                if value.shape != self.data_arr.data.shape:
+                    if value.ndim != 1:
+                        raise ValueError("1D weights expected when shapes of the data and weights differ.")
+                    if value.shape[0] != self.data_arr.data.shape[axis]:
```

Comment:
I think axis can be set to 0 here and removed as a parameter.  

There will need to be a test case for this added as well and then this should be ready